service/mcs/persistent/src/main/resources/create_cassandra_schema.cql

```diff
@@ -41,4 +41,11 @@ CREATE TABLE representation_versions (
 	creation_date timestamp,
 	PRIMARY KEY (cloud_id, schema_id, version_id)
 );
-CREATE INDEX representations_provider_id ON representation_versions (provider_id);
\ No newline at end of file
+CREATE INDEX representations_provider_id ON representation_versions (provider_id);
+
+CREATE TABLE data_sets_representations_names(

```

Comment:
From my point of view we agreed that eCloud should be able to answer questions mentioned in document. We didn't agree (we didn't discuss) about particular implementations. I also think that it is up to developer who implements sth, how it will be implemented. As long as solutions works (fulfills requirements) and is created with some rules (best practices) in mind, that is fine.
Moreover, there are still some bugs in this document, so I wouldn't trust it blindly. 

So why do I think that this approach is better:
1. This table is created to answer question: 'Give me all representation names that provider’s dataset has' so it is not a place for 'created_date' and 'updated_timestamp' for me. It is meaning-less in context of this table. But maybe I'm wrong, could You provide some use cases to use these fields?
2. According to Your use cases, number of representation names for given data set will be small. So this set will also be small (so it fallows general rules about sets introduced by Cassandra experts);
3. In this approach we have to read one row to answer question, in Your approach we have to read multiple rows and than merge them into some list or set.
4. For the question: 'check if a representation name is in a specific dataset': 
This is another one question that is not in document but still can be answered using 'my' implementation. Only one add on in this case will be: Set.contains(repName). From the performance point of view it is nothing.
5. In this one approach data is better distributed across nodes. In Your solution data sets created by one provider will go to the same partition, so some partitions will be much bigger than other (for providers that have many data sets);

I propose to switch this discussion on Slack (if there is still sth more to discuss)